dvc/remote/local.py

```diff
@@ -166,7 +166,9 @@ def move(self, from_info, to_info):
 
     @staticmethod
     def copy(from_info, to_info):
-        System.copy(from_info, to_info)
+        tmp_file = tmp_fname(to_info)
+        System.copy(from_info, tmp_file)
+        move(tmp_file, to_info)
```

Comment:
If we make this one atomic then proper abstraction requires us making `.copy()` atomic for all other remotes. And this will fix the same problem for SSH and HDFS external outputs)

Should we bundle that here? or create a separate ticket?

Reply:
By the way, there's already an `_atomic_copy` in SSH connection.

Reply:
Good point. Also, we don't seem to use atomic approach on `upload`(unlike `download`) so it would be nice to fix it there as well. Will look into it.

Reply:
Introduced atomic_copy (to not affect all `copy` instances, as I'm not sure that is needed).

Reply:
Ok, checked and it doesn't really hurt anything that much. The most harm we get is 2 renames when downloading from local remote, which is negligible.

Reply:
Ok, on further investigation I've remembered why it wasn't that way for all remotes from day one. The issue is that some remotes (s3, gs) don't have `rename` operation, so `move` is emulated by `copy + remove`, so atomic copy would not be possible, unless it is provided by the remote itself. So indeed, we only need to fix local, ssh and hdfs. Looking into that...